src/lib/storage/reference_segment.cpp

```diff
@@ -0,0 +1,36 @@
+#include <unordered_set>
+
+#include "reference_segment.hpp"
+#include "type_cast.hpp"
+
+#include "utils/assert.hpp"
+
+namespace opossum {
+
+ReferenceSegment::ReferenceSegment(const std::shared_ptr<const Table>& referenced_table,
+                                   const ColumnID referenced_column_id, const std::shared_ptr<const PosList>& pos)
+    : _referenced_table(referenced_table), _referenced_column_id(referenced_column_id), _pos(pos) {}
+
+AllTypeVariant ReferenceSegment::operator[](const ChunkOffset chunk_offset) const {
```

Comment:
even though the `[]` operator does not usually do bounds checking, a debug assert might still be helpful here to check if the chunk_offset is valid.